plugins/woocommerce/src/Internal/StockNotifications/Admin/NotificationsListTable.php

```diff
@@ -0,0 +1,799 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace Automattic\WooCommerce\Internal\StockNotifications\Admin;
+
+use Automattic\WooCommerce\Internal\DataStores\StockNotifications\StockNotificationsDataStore;
+use Automattic\WooCommerce\Internal\StockNotifications\Enums\NotificationStatus;
+use Automattic\WooCommerce\Internal\StockNotifications\Notification;
+
+/**
+ * Notifications list table for Customer Stock Notifications.
+ */
+class NotificationsListTable extends \WP_List_Table {
+
+	/**
+	 * Page URL.
+	 *
+	 * @const PAGE_URL
+	 */
+	const PAGE_URL = 'admin.php?page=customer_stock_notifications';
+
+	/**
+	 * Total view records.
+	 *
+	 * @var int
+	 */
+	public $total_items = 0;
+
+	/**
+	 * Total active records.
+	 *
+	 * @var int
+	 */
+	public $total_active_items = 0;
+
+	/**
+	 * Total pending records.
+	 *
+	 * @var int
+	 */
+	public $total_pending_items = 0;
+
+	/**
+	 * Total cancelled records.
+	 *
+	 * @var int
+	 */
+	public $total_cancelled_items = 0;
+
+	/**
+	 * Total sent records.
+	 *
+	 * @var int
+	 */
+	public $total_sent_items = 0;
+
+	/**
+	 * Are there any notifications in the DB?.
+	 *
+	 * @var int
+	 */
+	public $has_stock_notifications = false;
+
+	/**
+	 * Data store.
+	 *
+	 * @var StockNotificationsDataStore
+	 */
+	public $data_store;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		global $status, $page;
+
+		$this->data_store              = \WC_Data_Store::load( 'stock_notification' );
+		$this->total_items             = $this->data_store->query( array( 'return' => 'count' ) );
```

Comment:
These queries should happen in the prepare_items, no need to store them in props.